### ISSUE TITLE
Brak miniaturek

### ISSUE THREAD

**Issue by sindart** [Krukosz/Allegro_REST_RSS#8]
Pozwoliłem sobie przetestować skrypt i z przykrością musze stwierdzić, że nadal jest problem z mniaturkami - testowane na QuiteRSS i RSSOwl. Nie wiem może mój serwer coś ma źle ustawione bo przy próbie wpisania w przeglądarkę: "http://www.w***o.pl/Rss/image.php?url=https://a.allegroimg.com/original/032436/ee2f0ec74de982b982149e6eb27d" otrzymuje komunikat: "Forbidden You don't have permission to access /Rss/image.php on this server." Nie wiem czy da się to tak wyświetlić. Sam oryginalny obrazek z allegro jest wczytywany po wpisaniu w przeglądarkę => "https://a.allegroimg.com/original/032436/ee2f0ec74de982b982149e6eb27d". Wydaje mi się, że albo moje czytniki nie moga sobie poradzić z interpretacją obrazka albo skrypt od obrazków nie działa. Generalnie skrypt po ściągnięciu działa ok ale bez miniaturek obrazków. Wiecie może jak zdiagnozować w czym tkwi problem?

**Comment by Krukosz** [comment]
Skoro masz bład 403 przy ręcznym wpisywaniu powyższego adresu do przegladarki to jest coś nie tak ustawione. Czy to jest poprawna ścieżka do twojego skryptu z obrazkami?

**Comment by sindart** [comment]
Ale plik image.php jest w tej samej lokalizacji co index.php i zdaje się index.php działa bo generuje się wszystko...

**Comment by sindart** [comment]
Bo rozumiem, że "skrypt z obrazkami" masz na myśli plik image.php?

**Comment by Krukosz** [comment]
Tak, chodzi o image.php. Spróbuj wywołać plik bez żadnych argumentów - jeżeli tutaj pojawia się 403 to jest coś nie tak po stronie serwera.

**Comment by sindart** [comment]
Po wywowałniu http://www.w***o.pl/Rss/image.php bezposrednio w przegladrce pojawia się jedynie biała strona i nic się nie wyświetla więc zakładam, że po stronie serwera jest wszstko ok?

**Comment by sindart** [comment]
Tylko ten plik u mnie chyba nic nie robi skoro biała strona bez żadnego komunikatu?

**Comment by Krukosz** [comment]
A czy w folderze src masz plik SimpleImage.php? Tam jest klasa, która zajmuje się obrazkami i na podstawie której napisana jest funkcja do zmiany rozmiaru.

**Comment by sindart** [comment]
Tak dokładnie w głównych folderze w którym jest index.php oraz image.php znajduje się folder src a w tym folderze m.in. SimpleImage.php. Można to jakoś zdebugować dokąd dochodzi ok a odkąd coś jest nie tak?

**Comment by sindart** [comment]
Wszystko inne poza miniaturkami działa.

**Comment by Krukosz** [comment]
Dorzuć do image.php na samej górze linijki:
`ini_set('display_errors', 1);
error_reporting(E_ALL);`

I wtedy możesz debugować. Ja używam Firefoxa i `image.php` bez argumentów pokazuje mi błąd "obraz jest uszkodzony".

**Comment by sindart** [comment]
Wyświetla się ini_set('display_errors', 1); error_reporting(E_ALL); jak dam przed <?php ten wpis.

**Comment by sindart** [comment]
Jak dam po <?php to biała strona.

**Comment by sindart** [comment]
Jakieś pomysły dlaczego nie działają miniaturki?

**Comment by sindart** [comment]
Prośba o pomoc w rozwiązaniu problemu. Miniaturki nadal nie działają.

**Comment by Krukosz** [comment]
Jak dasz przed <?php to nic dziwnego, że nie działa, bo to musi być w tagach <?php.
Spróbuj dodać to przed "if" - linia 16.

Musisz uruchomić skrypt z jakimiś argumentami, bo bez argumentów wyświetla się pusta strona.

Jakiej przeglądarki używasz?
Jaką masz wersję php u siebie na serwerze? Być może nie masz doinstalowanej biblioteki do obsługi obrazków. Na jakim hostingu jesteś?
Zrób skrypt z phpinfo() i pokaż wynik.

**Comment by sindart** [comment]
Cześć temat mi się trochę odświeżył i wracam do problemu braku miniaturek, poniżej info o php konto na serwerze az.pl:
[phpinfo().pdf](https://github.com/Krukosz/Allegro_REST_RSS/files/4175102/phpinfo.pdf)

Prośba o pomoc, będę wdzięczny.

**Comment by Krukosz** [comment]
Cały czas masz tą białą stronę na image.php?
Sprawdź error_log serwera php, tam zapewne będziesz miał odpowiedź. Nie mam pojęcia, gdzie on może być, ponieważ w php_info nie ma żadnej informacji o tym.

**Comment by sindart** [comment]
Cześć,
tak nadal bialy ekran i nic, moze dalbym Ci dostep do ftp i popatrzylbys co
tam jest nie tak bo sie slabo orientuje w tym php. Wszystko chodzi oprocz
ministurek ktore bardzo sie przydaja...

W dniu niedz., 9.02.2020 o 13:23 Krukosz <notifications@github.com>
napisał(a):

**Comment by sindart** [comment]
Hey jeszcze raz prośba o pomoc, dałbyś radę pomóc mi w jakiś sposób zdiagnozować co się dzieje, że mój czytnik nie widzi miniaturek? A może sprawdziłbyś u siebie czy QuiteRSS wczytuje Ci miniaturki?

**Comment by Krukosz** [comment]
Podaj link do kanału, który utworzyłeś - sprawdzę to na swoim czytniku

**Comment by sindart** [comment]
Deleted

**Comment by Krukosz** [comment]
To nie wina czytnika RSS, tylko już po stronie serwera jest to generowane.
image.php zwraca błąd 403 - bez loga nic nie pomogę, bo nie wiem co się może dziać.

Nie mam pojęcia gdzie jest log błędów php w az.pl. U mnie, w linuxpl.com log jest w folderze logs obok folderu public_html.

Wg phpinfo() ścieżka root to: `/var/www/vhosts/26/120686/webspace/httpdocs`

Sprawdź czy w folderze webspace albo i wyżej (bo te 6 cyfr to Twój nr klienta) czy nie masz folderu logs.

**Comment by sindart** [comment]
generalnie zdjęcia na serwerze chodzą... chyba będę musiał zaczerpnąc pomocy az.pl lub założyć serwer na linuxpl.com.

**Comment by Krukosz** [comment]
Zdjęcia tak, ale skrypt image.php generuje błąd. Dlaczego? Bez logu się nie dowiemy.

**Comment by sindart** [comment]
A może coś dodać do tego skryptu, żeby wypluł błędy?

**Comment by Krukosz** [comment]
Jeżeli display errors, które zastosowałeś jakiś czas temu nie pokazuje nic to najprawdopodobniej globalne ustawienie serwera ukrywa błędy przed użytkownikiem.
Zaloguj sie na ftp i poszukaj pliku z logiem, bez tego nawet nie ma co zgadywać.

**Comment by sindart** [comment]
A może coś takie zastosować?
https://pomoc.home.pl/baza-wiedzy/jak-wlaczyc-logowanie-bledow-php-do-pliku

**Comment by Krukosz** [comment]
A nie łatwiej znaleźć plik log zamiast rzeźbić w skrypcie?

**Comment by sindart** [comment]
Szukam ale nie mogę nigdzie nic znaeźć może dam CI dane tymczasowe do ftpa i popatrzysz?

**Comment by Krukosz** [comment]
Podeślij mi dane na ******

**Comment by sindart** [comment]
Poszlo jak cos znajdziesz daj znac jesli nie to bede musial zasiegnac
pomocy az.pl dzieki za pomoc.

**Comment by Krukosz** [comment]
Dziwny ten hosting, zgłoś się do az.pl z pytaniem gdzie jest log. Dodałem ustawienia php.ini, które wymuszają prowadzenie loga, ale nic to nie dało.

**Comment by sindart** [comment]
Ok jutro zgłoszę do az.pl dzięki za pomoc dam znać jak czegoś się dowiem.
